app/models/provider_interface/provider_applications_page_state.rb

```diff
@@ -16,7 +16,32 @@ def filtered?
     end
 
     def applied_filters
-      @params.permit(:candidate_name, provider: [], status: [], accredited_provider: [], provider_location: []).to_h
+      @params.permit(:candidate_name, :sort_by, provider: [], status: [], accredited_provider: [], provider_location: []).to_h
+    end
+
+    def sort_by
+      sort_options.include?(@params[:sort_by]) ? @params[:sort_by] : 'Last changed'
+    end
+
+    def sort_options
+      ['Last changed', 'Days left to respond']
+    end
+
+    def sort_order
```

Comment:
Maybe for the future: at the cost of some indirection, we could move the actual sorting logic into a `SortApplicationChoicesForProviders` service to mirror `FilterApplicationChoicesForProviders` — `PageState` could go back to being ignorant of how the params are handled and give us just `last_changed` or `days_to_respond`

Reply:
I'll do this in this PR, I agree that there's a lot of sorty sort of stuff going on in '`PageState`

Reply:
@duncanjbrown https://github.com/DFE-Digital/apply-for-teacher-training/pull/2426/commits/fad8080ace092790a326319cbf93537de6b0c1cb I've delegated from sorting method calls to the sort service from page state. Not sure if this is a clean enough separation though, what do you think?

Reply:
That's definitely a cleaner separation. I think I'd go further — have page state know nothing at all about sorting, something like:

```
state = PageState.new(params, user)

application_choices = # query, filter, etc

application_choices = SortApplicationChoicesForProviders.call(
  application_choices: application_choices,
  sort_by: state.sort_by # returns :days_to_respond or :last_changed
)
```

i.e. `SortApplicationChoicesForProviders#call` would take the current `ApplicationChoices` relation and a `sort_by` and use a switch on the `sort_by` to select the sorting method and do the sorting at the same time, a bit like `FilterApplicationChoicesForProviders`.

Reply:
Ah! yes that's makes more sense I've amended the last commit https://github.com/DFE-Digital/apply-for-teacher-training/pull/2426/commits/ae9b93b0d4a6a4d0bca14aedf6f57b931b8fea66

Reply:
Nice! one last suggestion: https://github.com/DFE-Digital/apply-for-teacher-training/commit/71145b7e3a564c4503d8152608f9650b32e7a422 to make everything except `PageState` deal in `days_left_to_respond` and `last_changed`

Reply:
:+1: I've re-amended here https://github.com/DFE-Digital/apply-for-teacher-training/pull/2426/commits/dab9b2be664b2d0c7924ca5d4b0980636101e7b2 and included these in the option values which seemed to simplify things.